core/lib/zksync_core/src/native_token_fetcher/mod.rs

```diff
@@ -82,41 +113,45 @@ pub(crate) struct NativeTokenFetcher {
 }
 
 impl NativeTokenFetcher {
-    pub(crate) async fn new(config: NativeTokenFetcherConfig) -> Self {
-        let conversion_rate = reqwest::get(format!("{}/conversion_rate", config.host))
-            .await
-            .unwrap()
-            .json::<u64>()
-            .await
-            .unwrap();
-
+    pub(crate) async fn new(config: NativeTokenFetcherConfig) -> anyhow::Result<Self> {
         let http_client = reqwest::Client::new();
 
-        Self {
+        let conversion_rate = http_client
+            .get(format!("{}/conversion_rate", config.host))
+            .send()
+            .await?
+            .json::<u64>()
+            .await?;
+
+        Ok(Self {
             config,
             latest_to_eth_conversion_rate: AtomicU64::new(conversion_rate),
-            http_client: http_client,
-        }
+            http_client,
+        })
     }
 
     pub(crate) async fn run(&self, stop_receiver: watch::Receiver<bool>) -> anyhow::Result<()> {
+        let mut error_reporter = ErrorReporter::new();
         loop {
             if *stop_receiver.borrow() {
                 tracing::info!("Stop signal received, native_token_fetcher is shutting down");
                 break;
             }
 
-            let conversion_rate = self
+            match self
                 .http_client
                 .get(format!("{}/conversion_rate", &self.config.host))
                 .send()
-                .await?
-                .json::<u64>()
                 .await
-                .unwrap();
-
-            self.latest_to_eth_conversion_rate
-                .store(conversion_rate, std::sync::atomic::Ordering::Relaxed);
+            {
+                Ok(response) => {
+                    let conversion_rate = response.json::<u64>().await?;
```

Comment:
I guess it's fair to exit if the response is 200 but cannot be deserialized, but if done as implemented, the error will likely be pretty confusing and hard to analyze. Please add [`context`](https://docs.rs/anyhow/latest/anyhow/trait.Context.html#tymethod.context) here, e.g. `.await.context("Unable to parse the response of the native token conversion rate server")?`.

Reply:
Addressed this and comment bellow in latest commit.